tgui/packages/tgui/interfaces/Apc/index.js

```diff
@@ -0,0 +1,260 @@
+        return <font color="yellow">Low</font>;
+      case MAIN_STATUS_NONE:
+        return <font color="red">None</font>;
+    }
+  };
+
+  const chargingStatusToText = () => {
+    switch (charging) {
+      case 0:
+        return chargecount ? "Performing self-test" : "Not charging";
+      case 1:
+        return "Fully Charged";
+      default:
+        return "Charging";
+    }
+  };
+
+  const cellDisplay = () => {
+    if (cell_present) {
+      return (
+        <>
+          <LabeledList.Item label="Charging" direction="row">
+            <Button content="Off"
+              onClick={() => { onChargeModeChange(CHARGE_MODE_OFF); }}
+              disabled={!hasPermission() && (chargemode !== CHARGE_MODE_OFF)}
+              selected={chargemode === CHARGE_MODE_OFF}
+            />
+            <Button content="Auto"
+              onClick={() => { onChargeModeChange(CHARGE_MODE_AUTO); }}
+              disabled={!hasPermission() && (chargemode !== CHARGE_MODE_AUTO)}
+              selected={chargemode === CHARGE_MODE_AUTO}
+            />
+            <font>{"("}{chargingStatusToText()}{")"}</font>
+          </LabeledList.Item>
+          <LabeledList.Item label="Cell Power">
+            <ProgressBar value={cell_percent}
+              minValue={0}
+              maxValue={100}
+              color={cell_percent < 20 ? "red" : cell_percent < 50 ? "yellow" : "green"} />
+          </LabeledList.Item>
+        </>
+      );
+    } else {
+      return (
+        <LabeledList.Item label="Cell Power">
+          <ProgressBar value={cell_percent}
+            minValue={0}
+            maxValue={100}
+            color={cell_percent < 20 ? "red" : cell_percent < 50 ? "yellow" : "green"} />
+          <font color="red">Not Connected</font>
+        </LabeledList.Item>
+      );
+    }
+  };
+
+  const coverLockDisplay = () => {
+    let coverLockText = coverlocked ? "Engaged" : "Disengaged";
+    if (!hasPermission()) {
+      return <Box>{coverLockText}</Box>;
+    } else {
+      return <Button content={coverLockText} onClick={() => { onCoverLockedChange(!coverlocked); }} />;
+    }
+  };
+
+  const hasPermission = () => {
+    if (is_ai || is_silicon || can_access_remotely) {
+      return aidisabled ? false : true;
+    }
+    return locked ? false : true;
+  };
+
+  const isLocalAccess = () => {
+    return setup_networkapc < 2 && !can_access_remotely;
+  };
+
+  const hostConnectionDisplay = () => {
+    if (isLocalAccess()) {
+      return null;
+    } else {
+      return (
+        <LabeledList.Item label="Host Connection:">
+          <font color={host_connected ? "green" : "red"}>{host_connected ? "OK" : "NONE"}</font>
+        </LabeledList.Item>
+      );
+    }
+  };
+
+  const renderPoweredAreaApc = () => {
+    return (
+      <Window title="Area Power Controller" width={400} height={data["wiresexposed"] ? 680 : 420}>
+        <Window.Content>
+          <Section title={"Area Power Controller (" + area_name + ")"}>
```

Comment:
do we really need to say area power controller twice

Reply:
replaced with just the name of area in second instance